stake-pool/js/src/utils/stake.ts

```diff
@@ -71,55 +94,60 @@ export async function prepareWithdrawAccounts(
       });
     }
 
-    const transientStakeAccountAddress = await findTransientStakeProgramAddress(
-      STAKE_POOL_PROGRAM_ID,
-      validator.voteAccountAddress,
-      stakePoolAddress,
-      validator.transientSeedSuffixStart,
-    );
-
-    if (!validator.transientStakeLamports?.isZero()) {
+    const transientStakeLamports = validator.transientStakeLamports.toNumber() - minBalance;
+    if (transientStakeLamports > 0) {
+      const transientStakeAccountAddress = await findTransientStakeProgramAddress(
+        STAKE_POOL_PROGRAM_ID,
+        validator.voteAccountAddress,
+        stakePoolAddress,
+        validator.transientSeedSuffixStart,
+      );
       accounts.push({
         type: 'transient',
         voteAddress: validator.voteAccountAddress,
         stakeAddress: transientStakeAccountAddress,
-        lamports: validator.transientStakeLamports.toNumber(),
+        lamports: transientStakeLamports,
       });
     }
   }
 
   // Sort from highest to lowest balance
-  accounts = accounts.sort((a, b) => b.lamports - a.lamports);
+  accounts = accounts.sort(compareFn ? compareFn : (a, b) => b.lamports - a.lamports);
 
   const reserveStake = await connection.getAccountInfo(stakePool.reserveStake);
-  if (reserveStake && reserveStake.lamports > 0) {
-    console.log('Reserve Stake: ', reserveStake.lamports);
+  const reserveStakeBalance = (reserveStake?.lamports ?? 0) - minBalanceForRentExemption - 1;
+  if (reserveStakeBalance > 0) {
     accounts.push({
       type: 'reserve',
       stakeAddress: stakePool.reserveStake,
-      lamports: reserveStake?.lamports,
+      lamports: reserveStakeBalance,
     });
   }
 
   // Prepare the list of accounts to withdraw from
   const withdrawFrom: WithdrawAccount[] = [];
   let remainingAmount = amount;
 
+  const fee = stakePool.stakeWithdrawalFee;
+  const inverseFee: Fee = {
+    numerator: fee.denominator.sub(fee.numerator),
+    denominator: fee.denominator,
+  };
+
   for (const type of ['preferred', 'active', 'transient', 'reserve']) {
-    const filteredAccounts = accounts.filter(a => a.type == type);
+    const filteredAccounts = accounts.filter((a) => a.type == type);
 
-    for (const {stakeAddress, voteAddress, lamports} of filteredAccounts) {
-      let availableForWithdrawal = Math.floor(
-        calcPoolTokensForDeposit(stakePool, lamports),
-      );
-      if (!stakePool.stakeWithdrawalFee.denominator.isZero()) {
+    for (const { stakeAddress, voteAddress, lamports } of filteredAccounts) {
+      if (lamports <= minBalance) {
+        continue;
```

Comment:
Please change the above condition to `(lamports <= minBalance && type == "transient")` as the `minBalance` only applies to transient stake accounts. I believe that is all that needs to be fixed (besides `useReserve` which is a larger issue related to reserve accounts in general).